File: examples/project-template/src/main/java/be/kuleuven/csa/controller/ProjectMainController.java
package be.kuleuven.csa.controller;

import be.kuleuven.csa.ProjectMain;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.layout.AnchorPane;
import javafx.stage.Modality;
import javafx.stage.Stage;

import java.io.IOException;

public class ProjectMainController {

    @FXML
    private Button btnBoerderijen;
    @FXML
    private Button btnTips;
    @FXML
    private Button btnKlanten;
    @FXML
    private Button btnInschrijvingen;
    @FXML
    private Button btnProducten;

    public void initialize() {
        btnBoerderijen.setOnAction(e -> showBeheerScherm("boerderijen"));
        btnTips.setOnAction(e -> showBeheerScherm("tips"));
        btnKlanten.setOnAction(e -> showBeheerScherm("klanten"));
        btnInschrijvingen.setOnAction(e -> showBeheerScherm("inschrijvingen"));
        btnProducten.setOnAction(e -> showBeheerScherm("producten"));
    }

    private void showBeheerScherm(String id) {
        var resourceName = "beheer" + id + ".fxml";
        try {
            var stage = new Stage();
            var root = (AnchorPane) FXMLLoader.load(getClass().getClassLoader().getResource(resourceName));
            var scene = new Scene(root);
            stage.setScene(scene);
            stage.setTitle("Beheer van " + id);
            stage.initOwner(ProjectMain.getRootStage());
            stage.initModality(Modality.WINDOW_MODAL);
            stage.show();

        } catch (Exception e) {
            throw new RuntimeException("Kan beheerscherm " + resourceName + " niet vinden", e);
        }
    }
}

File: examples/kotlin/project-template/src/main/kotlin/be/kuleuven/csa/controller/ProjectMainController.kt
package be.kuleuven.csa.controller

import javafx.stage.Modality

import javafx.scene.Scene

import javafx.fxml.FXMLLoader

import javafx.scene.layout.AnchorPane

import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.stage.Stage
import java.lang.Exception

import be.kuleuven.csa.rootStage

class ProjectMainController {
    @FXML
    private lateinit var btnBoerderijen: Button

    @FXML
    private lateinit var btnTips: Button

    @FXML
    private lateinit var btnKlanten: Button

    @FXML
    private lateinit var btnInschrijvingen: Button

    @FXML
    private lateinit var btnProducten: Button

    fun initialize() {
        btnBoerderijen.setOnAction { e -> showBeheerScherm("boerderijen") }
        btnTips.setOnAction { e -> showBeheerScherm("tips") }
        btnKlanten.setOnAction { e -> showBeheerScherm("klanten") }
        btnInschrijvingen.setOnAction { e -> showBeheerScherm("inschrijvingen") }
        btnProducten.setOnAction { e -> showBeheerScherm("producten") }
    }

    private fun showBeheerScherm(id: String) {
        val resourceName = "/beheer$id.fxml"
        try {
            val stage = Stage()
            val root = FXMLLoader.load<Any>(this::class.java.getResource(resourceName)) as AnchorPane
            val scene = Scene(root)
            stage.setScene(scene)
            stage.setTitle("Beheer van $id")
            stage.initOwner(rootStage)
            stage.initModality(Modality.WINDOW_MODAL)
            stage.show()
        } catch (e: Exception) {
            throw RuntimeException("Kan beheerscherm $resourceName niet vinden", e)
        }
    }
}